src/hotspot/share/cds/archiveBuilder.hpp

```diff
@@ -126,15 +126,17 @@ class ArchiveBuilder : public StackObj {
     uintx _ptrmap_start;     // The bit-offset of the start of this object (inclusive)
     uintx _ptrmap_end;       // The bit-offset of the end   of this object (exclusive)
     bool _read_only;
+    bool _has_embedded_pointer;
     FollowMode _follow_mode;
     int _size_in_bytes;
+    int _index;              // The location of this object in _source_objs
```

Comment:
After sorting, this is no longer the location of this object in _source_objs. How about:

```
int _id; // Each object has a unique serial ID, starting from zero. The ID is assigned
         // when the object is added into _source_objs.
```